packages/core/src/providers/atlassian.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * <div style={{display: "flex", justifyContent: "space-between", alignItems: "center"}}>
+ * <span style={{fontSize: "1.35rem" }}>
+ *  Built-in sign in with <b>Atlassian</b> integration.
+ * </span>
+ * <a href="https://www.atlassian.com/" style={{backgroundColor: "black", padding: "12px", borderRadius: "100%" }}>
+ *   <img style={{display: "block"}} src="https://authjs.dev/img/providers/atlassian.svg" width="24" style={{ marginTop: "-3px"}} />
+ * </a>
+ * </div>
+ *
+ * @module providers/atlassian
+ */
+import type { OAuthConfig, OAuthUserConfig } from "./index.js"
+
+/** The returned user profile from Atlassian when using the profile callback. */
+export interface AtlassianProfile extends Record<string, any> {
+  /**
+   * The user's atlassian account ID
+   */
+  account_id: string
+  /**
+   * The user name
+   */
+  name: string
+  /**
+   * The user's email
+   */
+  email: string
+  /**
+   * The user's profile picture
+   */
+  picture: string
+}
+
+// Any config required that isn't part of the `OAuthUserConfig` spec should belong here
+// For example, we must pass a `redirectUri` to the Atlassian API when requesting tokens, therefore we add it here
+interface AdditionalConfig {
+  redirectUri: string | undefined
+}
+
+/**
+ * ### Setup
+ *
+ * #### Callback URL
+ * ```
+ * https://example.com/api/auth/callback/atlassian
+ * ```
+ *
+ * #### Configuration
+ *
+ * Import the provider and configure it in your **Auth.js** initialization file:
+ *
+ * ```ts title="pages/api/auth/[...nextauth].ts"
+ * import NextAuth from "next-auth"
+ * import AtlassianProvider from "next-auth/providers/atlassian"
+ *
+ * export default NextAuth({
+ *   providers: [
+ *     AtlassianProvider({
+ *       clientId: process.env.ATLASSIAN_ID,
+ *       clientSecret: process.env.ATLASSIAN_SECRET,
+ *       redirectUri: process.env.ATLASSIAN_REDIRECT_URI,
+ *     }),
+ *   ],
+ * })
+ * ```
```

Comment:
```suggestion
 * ```ts
 * import Atlassian from "@auth/core/providers/atlassian"
 * ...
 * providers: [
 *  Atlassian({
 *    clientId: env.AUTH_ATLASSIAN_ID,
 *    clientSecret: env.AUTH_ATLASSIAN_SECRET,
 *  }),
 * ]
 * ...
 * ```
```

Let's use a framework-agnostic example here.